gunshot/data-processing-application/CSV.java

```diff
@@ -0,0 +1,108 @@
+import java.io.File;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+/**
+ * CSV Class
+ * 
+ * <p>CSV is a class for the reading of a dataset in a csv format
+ * 
+ * @author Carver Forbes
+ * @version 0.1
+ */
+public class CSV {
+  private File file;
+
+  /**
+   * CSV constructor
+   * 
+   * @param pathname the pathname for the csv file to be read
+   * @throws IllegalArgumentException if the pathname is incorrect
+   */
+  public CSV(String pathname) throws IllegalArgumentException {
+    file = new File(pathname);
+    //TODO: handle permission issues
+    if (!file.exists()) {
+        throw new IllegalArgumentException("No such file exists");
+    }
+  }   
+
+  /** Tester Function to print 5 YouTube Links */
+  public void printYouTubeLinks() {
+    // prints youtube links
+    try {
+      Scanner scan = new Scanner(file);
+      String line = scan.nextLine();
+      while (line.charAt(0) == '#') {
+        line = scan.nextLine();
+      }
+      // while (scan.hasNext()) {
+      //   String ytid = line.split(",")[0];
+      //   String url = "https://youtube.com/watch?v=" + ytid;
+      //   System.out.println(url);
+      //   line = scan.nextLine();
+      // }
+      for (int i = 0; i < 5; i++) {
+        String ytid = line.split(",")[0];
+        String url = "https://youtube.com/watch?v=" + ytid;
+        System.out.println(url);
+        line = scan.nextLine();
+      }
+      scan.close();
```

Comment:
If an exception is thrown during this try block, the scanner will not be properly closed. Make sure to use a `finally` block when working with resource management in Java. Link: https://docs.oracle.com/javase/tutorial/essential/exceptions/finally.html.